addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/report/AuthReportData.java

```diff
@@ -32,13 +33,44 @@
 import lombok.Setter;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
 import org.zaproxy.addon.authhelper.internal.db.Diagnostic;
 import org.zaproxy.addon.authhelper.internal.db.TableJdo;
 
 @Getter
 @Setter
 public class AuthReportData implements Closeable {
 
+    @Getter
+    public enum FailureDetail {
+        OVERALL(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.overall.failed")),
+        PASS_COUNT(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.pass.count.failed")),
+        SESSION_MGMT(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.sessmgmt.failed")),
+        STATS_FAILURE(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.stats.failure.failed")),
+        STATS_SUCCESS(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.stats.success.failed")),
+        VERIF_IDENT(
+                Constant.messages.getString(
+                        "authhelper.authreport.summary.fail.detail.verif.failed"));
+
+        private final String key;
+        private final String detail;
+
+        private FailureDetail(String detail) {
+            key = name().toLowerCase(Locale.ROOT);
```

Comment:
The key should probably be something like `auth.failure.<name>` to fit in with the existing success keys.